laniakea/core/providers/azure/manager.py

```diff
@@ -2,7 +2,25 @@
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
+import re
+import sys
+import json
+import random
+import os.path
 import logging
+import argparse
+import unicodedata
+import appdirs
+
+try:
+    import azure
+except ImportError:
+    import pip
+    pip.main(['install', 'azure'])
```

Comment:
@rforbes I guess we should surround the pip.main with an additional try/except and in the except block: sys.exit("Unable to install pip")
@jschwartzentruber do you have any recommendations here? I wanted not to pollute the requirements.txt with module specific libraries and azure is quite huge.